resources/assets/lib/user-card.tsx

```diff
@@ -176,7 +177,7 @@ export class UserCard extends React.PureComponent<Props, State> {
       background = <div className={overlayCssClass} />;
     }
 
-    if (this.isUserLoaded) {
+    if (this.isUserLoaded && !this.user.is_deleted) {
```

Comment:
this combination can be made `isUserVisible` property or something, or just have different render for deleted user.